cpp/cg_method/_cg_method.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef __CG_METHOD_HPP__
+#define __CG_METHOD_HPP__
+
+#include <vector>
+#include "../matrix/_matrix.hpp"
+#include <algorithm>
+#include <cmath>
+
```

Comment:
Most of the time we want to use a namespace for C++ code.  If you do not intend to maintain the code longer than 2 months, you don't have to use it, though.

Reply:
Sorry, I don't get it. 
Do you mean that I shouldn't use ```#include "..."``` in the header?

Reply:
No.  I meant to use namespace like https://github.com/solvcon/modmesh/blob/4c1e00fa47fc353798e739fee736cf00740b6855/cpp/modmesh/buffer/SimpleArray.hpp#L41